File: app/src/main/java/org/kiwix/kiwixmobile/data/AbstractContentProvider.java
package org.kiwix.kiwixmobile.data;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

/*
 * Kiwix Android
 * Copyright (C) 2018  Kiwix <android.kiwix.org>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
abstract class AbstractContentProvider extends ContentProvider {
  @Override
  public Cursor query(Uri url, String[] projection, String selection,
    String[] selectionArgs, String sort) {
    throw new RuntimeException("Operation not supported");
  }

  @Override
  public Uri insert(Uri uri, ContentValues initialValues) {
    throw new RuntimeException("Operation not supported");
  }

  @Override
  public int update(Uri uri, ContentValues values, String where,
    String[] whereArgs) {
    throw new RuntimeException("Operation not supported");
  }

  @Override
  public int delete(Uri uri, String where, String[] whereArgs) {
    throw new RuntimeException("Operation not supported");
  }
}

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/AbstractContentProvider.kt
/*
 * Kiwix Android
 * Copyright (c) 2019 Kiwix <android.kiwix.org>
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.kiwix.kiwixmobile.core.data

import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri

internal abstract class AbstractContentProvider : ContentProvider() {
  override fun query(
    url: Uri,
    projection: Array<String>?,
    selection: String?,
    selectionArgs: Array<String>?,
    sort: String?
  ): Cursor? {
    throw RuntimeException("Operation not supported")
  }

  override fun insert(uri: Uri, initialValues: ContentValues?): Uri? {
    throw RuntimeException("Operation not supported")
  }

  override fun update(
    uri: Uri,
    values: ContentValues?,
    where: String?,
    whereArgs: Array<String>?
  ): Int {
    throw RuntimeException("Operation not supported")
  }

  override fun delete(uri: Uri, where: String?, whereArgs: Array<String>?): Int {
    throw RuntimeException("Operation not supported")
  }
}
